py/desisim/scripts/quickquasars.py

```diff
@@ -245,7 +283,12 @@ def simulate_one_healpix(ifilename,args,model,obsconditions,decam_and_wise_filte
         nmodel=nqso, redshift=metadata['Z'], 
         lyaforest=False, nocolorcuts=True, noresample=True, seed = seed)
 
+    ##To add BALs to be checked by Luz and Jaime
+    if args.balprob:
+       bal=BAL()
+       tmp_qso_flux,meta_bal=bal.insert_bals(tmp_qso_wave,tmp_qso_flux, metadata['Z'], balprob=args.balprob)
 
```

Comment:
You should pass `seed=seed` to `bal.insert_bals()` on line 289 to ensure reproducibility (i.e., that the same QSOs become BAL QSOs when the seed is the same).

Also consider passing the `BAL()` object as an argument to `simulate_one_healpix` so you don't read the BAL basis templates over and over.